### PR TITLE
used spotlight package in dockerfile instead of just sidecar to serve index file

### PR DESCRIPTION
**Pull request by Shubhdeep12**
… index page.

Before opening this PR:
* [x] I added a [Changeset Entry](https://spotlightjs.com/contribute/changesets/) with `pnpm changeset:add`
* [ ] I referenced issues that this PR addresses

### CODE REVIEW

File: packages/spotlight/bin/run.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env node
 const { setupSidecar } = await import('../dist/sidecar.js');
-const port = process.argv[2];
+const port = process.argv.length >= 3 ? Number(process.argv[2]) : undefined;
```
Comment (Lms24) [inline packages/spotlight/bin/run.js:3]: good catch!

**Review by Lms24 — APPROVED** [review]

**Review by HazAT — APPROVED** [review]
💯

**Comment by Shubhdeep12** [comment]
This will fail the docker test
seems like we have to build everything - overlay, sidecar and then spotlight to make this work

**Comment by HazAT** [comment]
> This will fail the docker test seems like we have to build everything - overlay, sidecar and then spotlight to make this work

Right makes sense, we should build everything

File: packages/spotlight/bin/run.js (hunk 1 above)
Comment (Shubhdeep12) [reply packages/spotlight/bin/run.js:3]: yes, but it was done by @dcramer initially in another PR. I thought its better to add it here as well.

**Comment by Shubhdeep12** [comment]
I think now it will work :)

File: packages/spotlight/bin/run.js (hunk 1 above)
Comment (dcramer) [reply packages/spotlight/bin/run.js:3]: would be great if we could figure out how to dedupe this script, but this is beyond my domain knowledge w/ npm packages

**Review by dcramer — APPROVED** [review]

**Pull request merged**